### PR TITLE
Implement the Run, Halt, Step, Load and Store operations

### PR DESCRIPTION
**Pull request by toon-leader-bacon**

### CODE REVIEW

File: src/main/java/group1/mips_simulator/SimController.java [hunk 1]
```diff
@@ -69,41 +71,144 @@ protected void restartComputer() {
...
     }
 
+    /**
+     * Similar to the loadOnClick function (read value from memory pointed
+     * to by MAR into MBR) but this will also increase the MAR AFTER
+     * the read.
+     */
     public void loadPlusOnClick(ActionEvent actionEvent) {
+        loadOnClick(actionEvent);
+        Register memAddrReg = this.computer.cpu.regfile.getMAR();
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/SimController.java:124]: minor adjustment, non-blocker
consistent naming of memAddrReg vs memAddReg

personally like memAddrReg, but memAddReg fine too and requires fewer modifications

File: src/main/java/group1/mips_simulator/SimController.java [hunk 2]
```diff
@@ -69,41 +71,144 @@ protected void restartComputer() {
...
+        userHalt = true;
+        if (programRunnerThread != null) {
+            programRunnerThread.interrupt();
+        }
+
+        System.out.println("Halt Redrawing");
+        // redraw the frame after forcing the program to stop early to render the
+        // changes in the Model to the View.
+        redraw();
+        System.out.println("Finished HALTING!!!!!!!!!!!!!!!!!!!!!!!");
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/SimController.java:211]: very intense, nice

File: src/main/java/group1/mips_simulator/Utility.java [hunk 3]
```diff
@@ -97,11 +97,26 @@ public static String binaryStrToOctalStr(String binaryStr) {
...
+    public static String shortToBinaryString(short value, int minSize) {
         StringBuilder sb = new StringBuilder(Integer.toBinaryString(0xFFFF & value));
         while (sb.length() < minSize) {
             sb.insert(0, '0');
         }
+        return sb.toString();
+    }
+
+    public static String shortToBinaryString_Pretty(short value, int minSize) {
+        int bitMask = 0b1111_1111_1111_1111;
```
Comment (akm303) [inline src/main/java/group1/mips_simulator/Utility.java:109]: not a real note, just didnt know you could use '_' in your bit-ints, thats pretty cool

**Review by akm303 — APPROVED** [review]
no glaring bugs, also tried it on my own and it works like a charm; really really excellent stuff man
damn and here i was just trying to figure how to get the rom loading working...

feel free to push through whenever you feel is right, left one note about a consistent-naming thing, but its super minor

File: src/main/java/group1/mips_simulator/SimController.java (hunk 1 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/SimController.java:124]: Just so I'm clear, you're asking that I remove the `r` small-r character from `memAddrReg`? Will update.

Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/SimController.java:124]: Also, a pro tip for GitHub. When you leave a comment there's a button that looks like a page with a + - character on it called "Suggest a Change" 
image

If you click that, you can make a change to a single line and make a cool interface to allow the merge author to make the change with a single button click

File: src/main/java/group1/mips_simulator/SimController.java (hunk 2 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/SimController.java:211]: Well, when it runs 5m instructions per second I was afraid that I'd miss this output.

File: src/main/java/group1/mips_simulator/Utility.java (hunk 3 above)
Comment (toon-leader-bacon) [reply src/main/java/group1/mips_simulator/Utility.java:109]: You can use them in regular `int` and when you're defining hex literals `0xFFF_FFF_FFF` for example too

**Pull request merged**